libdrgn/configure.ac

```diff
@@ -37,6 +37,24 @@ AS_CASE(["x$enable_openmp"],
 AC_SUBST(OPENMP_CFLAGS)
 AC_SUBST(OPENMP_LIBS)
 
+AC_ARG_ENABLE([asan],
+	      [AS_HELP_STRING([--enable-asan@<:@=ARG@:>@],
+			      [use ASAN. ARG may be yes or no.
+			       @<:@default=no@:>@])],
+	      [], [enable_asan=no])
+
+ASAN_CFLAGS=
+ASAN_LIBS=
+ASAN_LDFLAGS=
+AS_CASE(["x$enable_asan"],
+	[xyes], [ASAN_CFLAGS="-fsanitize=address -fno-omit-frame-pointer"
+			 ASAN_LIBS=-lasan
```

Comment:
From what I understand, you're not supposed to link against libasan directly. See https://github.com/google/sanitizers/issues/796, for example. Instead, it seems like the proper way to use ASAN for a shared library is with `LD_PRELOAD`. This is how I've used ASAN (on Arch Linux, the location of `libasan.so` may vary on other distros) in the past:

```console
$ CFLAGS="-Wall -Werror -g -O2 -fsanitize=address -fno-omit-frame-pointer" LDFLAGS="-fsanitize=address" python3 setup.py build_ext -i
$ LD_PRELOAD="/lib/libasan.so" python3 -Bm unittest discover -v .
```

Reply:
I've also been running it with `LD_PRELOAD`, and I could've sworn it wasn't working without `-lasan`, but upon re-testing it appears to be unnecessary, will remove it.